File: BFS/1260.c
#include <stdio.h>
#include <string.h>

int graph[1001][1001];
int visit[1001];
int queue[1001];

void dfs(int n, int v)
{
	visit[v] = 1;

	printf("%d ", v);

	for (int i = 1; i <= n; i++) {
		if (visit[i] == 0 && graph[v][i] == 1) {
			dfs(n, i);
		}
	}

	return;
}

void bfs(int n, int v)
{
	int front = -1, rear = -1, pop;
	printf("%d ", v);
	visit[v] = 1;
	queue[++rear] = v;

	while (front < rear) {
		pop = queue[++front];
		for (int i = 1; i <= n; i++) {
			if (visit[i] == 0 && graph[pop][i] == 1) {
				printf("%d ", i);
				visit[i] = 1;
				queue[++rear] = i;
			}
		}
	}
}

int main()
{
	int n, m, v;
	int x, y;
	scanf("%d %d %d", &n, &m, &v);
	
	for (int i = 0; i < m; i++) {
		scanf("%d %d", &x, &y);
		graph[x][y] = 1;
		graph[y][x] = 1;
	}

	dfs(n, v);
	printf("\n");
	memset(visit, 0, sizeof(int) * (n + 1));
	bfs(n, v);
}

File: BFS/1260.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(100000)
from  collections import deque

n, m, v = map(int, input().split())

nodes = [[] for _ in range(n + 1)]
graph = [[0] * (n + 1) for _ in range(n + 1)]
visit = [False for _ in range(n + 1)]
visit_b = [False for _ in range(n + 1)]

for i in range(m):
    x, y = map(int, input().split())
    nodes[x].append(y)
    nodes[y].append(x)
    graph[x][y] = 1
    graph[y][x] = 1


for i in range(n + 1):
    nodes[i].sort()

def dfs(node):
    visit[node] = True
    print(node, end=' ')

    for i in nodes[node]:
        if visit[i] == False:
            dfs(i)

def bfs(node):
    q = deque([node])
    visit_b[node] = True

    while q:
        node = q.popleft()
        print(node, end=' ')
        for i in range(1, n + 1):
            if visit_b[i] == False and graph[node][i] == 1:
                q.append(i)
                visit_b[i] = True

dfs(v)
print()
bfs(v)
